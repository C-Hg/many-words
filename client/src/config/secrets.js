module.exports = {
  GOOGLE_CLIENT_ID: process.env.REACT_APP_GOOGLE_CLIENT_ID,
  FACEBOOK_APP_ID: process.env.REACT_APP_FACEBOOK_APP_ID
};
