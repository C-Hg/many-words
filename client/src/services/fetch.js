import { fetch } from "whatwg-fetch";

const services = {
  async socialLogin(provider, token) {
    const options = {
      method: "get",
      headers: {
        access_token: token,
        "Content-type": "text/plain"
      }
    };
    const result = await fetch(`/auth/${provider}/token`, options);
    return result.json();
  },

  async getJSONResponse(url) {
    const result = await fetch(url);
    return result.json();
  },

  async getTextResponse(url) {
    const result = await fetch(url);
    return result.text();
  },

  async postJSONResponse(url, data) {
    const options = {
      method: "post",
      headers: {
        "Content-Type": "application/json"
      },
      body: data
    };
    const result = await fetch(url, options);
    return result.json();
  }
};

export default services;
