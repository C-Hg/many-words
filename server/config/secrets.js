//placeholder allow compiling without app secrets for social auth

module.exports = {
  SESSION_SECRET: process.env.SESSION_SECRET || "session_secret",

  GOOGLE_CLIENT_ID: process.env.GOOGLE_CLIENT_ID || "no_ID",
  GOOGLE_CLIENT_SECRET: process.env.GOOGLE_CLIENT_SECRET || "no_secret",

  FACEBOOK_APP_ID: process.env.FACEBOOK_APP_ID || "no_ID",
  FACEBOOK_APP_SECRET: process.env.FACEBOOK_APP_SECRET || "no_secret",

  MONGO_URI: "mongodb://localhost/many-words"
};
