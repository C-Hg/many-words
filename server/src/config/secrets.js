// placeholders allow compiling without app secrets for social auth
const {
  NODE_ENV,
  MONGODB_PASSWORD,
  MONGODB_USER,
  MONGO_INITDB_DATABASE,
  GOOGLE_CLIENT_ID,
  GOOGLE_CLIENT_SECRET,
  SESSION_SECRET
} = process.env;

module.exports = {
  SESSION_SECRET: SESSION_SECRET || "session_secret",

  GOOGLE_CLIENT_ID: GOOGLE_CLIENT_ID || "no_ID",
  GOOGLE_CLIENT_SECRET: GOOGLE_CLIENT_SECRET || "no_secret",

  FACEBOOK_APP_ID: process.env.FACEBOOK_APP_ID || "no_ID",
  FACEBOOK_APP_SECRET: process.env.FACEBOOK_APP_SECRET || "no_secret",

  // mongo is the name of the docker image
  MONGO_URI:
    NODE_ENV === "production"
      ? `mongodb://${MONGODB_USER}:${MONGODB_PASSWORD}@mongo:27017/${MONGO_INITDB_DATABASE}`
      : "mongodb://mongo:27017/many-words"
};
