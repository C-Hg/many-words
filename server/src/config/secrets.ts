// placeholders allow compiling without app secrets for social auth
const {
  JWT_SECRET,
  MONGODB_PASSWORD,
  MONGODB_USER,
  MONGO_INITDB_DATABASE,
  NODE_ENV,
  SERVER_PORT,
} = process.env;

const secrets = {
  JWT_SECRET,
  NODE_ENV,
  SERVER_PORT,

  // mongo is the name of the docker image
  MONGO_URI:
    NODE_ENV === "production"
      ? `mongodb://${MONGODB_USER}:${MONGODB_PASSWORD}@mongo:27017/${MONGO_INITDB_DATABASE}`
      : "mongodb://mongo:27017/many-words",
};

export default secrets;
