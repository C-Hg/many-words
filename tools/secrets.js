const {
  NODE_ENV,
  MONGODB_PASSWORD,
  MONGODB_USER,
  MONGO_INITDB_DATABASE
} = process.env;

module.exports = {
  MONGO_URI:
    NODE_ENV === "production"
      ? `mongodb://${MONGODB_USER}:${MONGODB_PASSWORD}@mongo:27017/${MONGO_INITDB_DATABASE}`
      : "mongodb://mongo:27017/many-words"
};
