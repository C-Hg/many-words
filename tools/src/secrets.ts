const {
  NODE_ENV,
  MONGODB_PASSWORD,
  MONGODB_USER,
  MONGO_INITDB_DATABASE,
} = process.env;

const secrets = {
  MONGO_URI:
    NODE_ENV === "production"
      ? `mongodb://${MONGODB_USER}:${MONGODB_PASSWORD}@mongo:27017/${MONGO_INITDB_DATABASE}`
      : "mongodb://mongo:27017/many-words",
};

export default secrets;
