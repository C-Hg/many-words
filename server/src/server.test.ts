import { toPromise, execute } from "apollo-link";
import { HttpLink } from "apollo-link-http";
import gql from "graphql-tag";
import fetch from "node-fetch";

import secrets from "./config/secrets";

const USER_QUERY = gql`
  query user {
    user {
      id
      email
    }
  }
`;

// TODO: manage JWT https://blog.logrocket.com/writing-end-to-end-tests-for-graphql-servers-using-jest/
// TODO: docker with watch to use secrets, with different ports
describe("Server - e2e", () => {
  // --------------------->
  const link = new HttpLink({
    uri: `http://localhost:4000/graphql`,
    fetch,
    // headers: {
    //   authorization:
    // }
  });

  const graphql = ({ query, variables = {} }) =>
    execute(link, { query, variables });
  // <------------------------- get out in a file to call graphql with or without auth
  // and tests e2e next to the schemas
  // next to server : test auth or not

  it.only("gets user", async () => {
    const res = await toPromise(
      graphql({
        query: USER_QUERY,
      })
    );

    expect(res).toMatchSnapshot();
  });
});
