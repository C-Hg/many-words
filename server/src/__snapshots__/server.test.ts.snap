// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e should get user 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "camille.hagenbourger@gmail.com",
      "id": "5d66dc6a8946c00184ab1102",
    },
  },
}
`;
